src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -353,18 +411,25 @@ define([
       });
     },
 
-    showAppModal: function(app) {
+    routeApp: function(params) {
       if (this.state.modalClass !== null) {
         return;
       }
 
+      var app = this.state.collection.get("/"+params[0]);
+
+      if(!app) {
+        window.location.hash = "#";
+        return;
+      }
+
       this.setState({
```

Comment:
Should not set state in route function. Do this in `componentDidMount`, i.e. before render. This should only result in a component.